### PR TITLE
#58: Implement getHostIp()

### LINKED ISSUE #58: Implement getHostIp()
```
java.lang.UnsupportedOperationException: getTestHostIpAddress() is only implemented for docker-machine right now

	at org.testcontainers.containers.GenericContainer.getTestHostIpAddress(GenericContainer.java:1297)
	at com.exasol.adapter.dynamodb.ExasolTestInterface.getTestHostIpAddress(ExasolTestInterface.java:173)
	at com.exasol.adapter.dynamodb.ExasolTestInterface.setScriptOutputAddress(ExasolTestInterface.java:143)
	at com.exasol.adapter.dynamodb.ExasolTestInterface.createAdapterScript(ExasolTestInterface.java:116)
	at com.exasol.adapter.dynamodb.DynamodbAdapterTestLocalIT.beforeAll(DynamodbAdapterTestLocalIT.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:628)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:117)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$invokeBeforeAllMethods$9(ClassTestDescriptor.java:376)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.invokeBeforeAllMethods(ClassTestDescriptor.java:375)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.before(ClassTestDescriptor.java:201)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.before(ClassTestDescriptor.java:77)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
```

### LINKED ISSUE #122: Build on github fails due to old testcontainers version
Github actions for repositories that use ETC fail, with:
```
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=exasol/docker-db:7.0.4, imagePullPolicy=DefaultPullPolicy())
Caused by: com.github.dockerjava.api.exception.NotFoundException: 
Status 404: {"message":"No such image: testcontainers/ryuk:0.3.0"}
```
See: https://github.com/testcontainers/testcontainers-java/issues/3574

Fix: Update test containers to 1.15.1

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #58.
Updated test container basis. Closes #122.

### CODE REVIEW

File: doc/changes/changelog.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Changes
 
+* [3.4.1](changes_3.5.0.md)
```
Comment (AnastasiiaSergienko) [inline doc/changes/changelog.md:3]:
```suggestion
* [3.5.0](changes_3.5.0.md)
```

File: doc/changes/changes_3.5.0.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+In the 3.5.0 release of the Exasol Test Containers, we added a new function `getHostIp()` that allows getting the IP
+address of the host the Exasol docker container runs on. This is useful if you want to connect to a debugger or profiler.
+
+Note that this feature is at the moment only available on Linux.
+
+## Features
+
+* #68: Support `getHostIp()` on Linux
+
+With release 3.5.0 the Exasol Test Container now provides the function `getHostIp()` that lets you acquire the IP address of the host that runs the Exasol `docker-db`. This is for example useful if you have a debugger or profiler running on the host and want to connect to it from inside the Docker container.
```
Comment (AnastasiiaSergienko) [inline doc/changes/changes_3.5.0.md:16]: It's a duplication of summary

File: doc/changes/changes_3.5.0.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+We also updated the default `docker-db` image version to 7.0.6.
```
Comment (AnastasiiaSergienko) [inline doc/changes/changes_3.5.0.md:18]: This should go to summary

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 4]
```diff
@@ -704,4 +704,14 @@ public final synchronized ExasolDriverManager getDriverManager() {
...
         return this.driverManager;
     }
+
+    /**
+     * Get the IP address of the host running this container.
+     *
+     * @return IP address of the host
+     */
+    public String getHostIp() {
+        final HostIpDetector detector = new HostIpDetector(this);
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/containers/ExasolContainer.java:714]: What if we create an instance of HostIpDetector once in the constructor and then use `getHostIp(this)` method?

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]
A few small comment

File: doc/changes/changelog.md (hunk 1 above)
Comment (redcatbear) [reply doc/changes/changelog.md:3]: Fixed.

File: doc/changes/changes_3.5.0.md (hunk 2 above)
Comment (redcatbear) [reply doc/changes/changes_3.5.0.md]: Fixed.

File: doc/changes/changes_3.5.0.md (hunk 3 above)
Comment (redcatbear) [reply doc/changes/changes_3.5.0.md]: Fixed.

File: src/main/java/com/exasol/containers/ExasolContainer.java (hunk 4 above)
Comment (redcatbear) [reply src/main/java/com/exasol/containers/ExasolContainer.java:714]: Also possible. I thought about whether that is necessary, but the method is seldom used and the detector has only one instance variable that needs to be initialized with a reference, so it is really cheap. In this case encapsulation wins over reuse-optimization.

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**